dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -169,28 +166,43 @@ protected PagedDoraWorker(
     super(ExecutorServiceFactories.fixedThreadPool("dora-worker-executor", 5));
     mWorkerId = workerId;
     mConf = conf;
-    mRootUFS = Configuration.getString(PropertyKey.DORA_CLIENT_UFS_ROOT);
+    mRootUFS = mConf.getString(PropertyKey.DORA_CLIENT_UFS_ROOT);
     mUfsManager = mResourceCloser.register(new DoraUfsManager());
+    mUfsManager.getOrAdd(new AlluxioURI(mRootUFS),
+        UnderFileSystemConfiguration.defaults(mConf));
     mFsContext = mResourceCloser.register(fileSystemContext);
     mUfsStreamCache = new UfsInputStreamCache();
-    mUfs = UnderFileSystem.Factory.create(
-        mRootUFS,
-        UnderFileSystemConfiguration.defaults(Configuration.global()));
 
-    mPageSize = Configuration.global().getBytes(PropertyKey.WORKER_PAGE_STORE_PAGE_SIZE);
+    mPageSize = mConf.getBytes(PropertyKey.WORKER_PAGE_STORE_PAGE_SIZE);
     mBlockMasterClientPool = blockMasterClientPool;
     mCacheManager = cacheManager;
     mMetaManager = mResourceCloser.register(
-        new DoraMetaManager(this, mCacheManager, mUfs));
+        new DoraMetaManager(mConf, this, mCacheManager, mUfsManager));
     mOpenFileHandleContainer = new DoraOpenFileHandleContainer();
 
     mMkdirsRecursive = MkdirsOptions.defaults(mConf).setCreateParent(true);
     mMkdirsNonRecursive = MkdirsOptions.defaults(mConf).setCreateParent(false);
 
-    mClientWriteToUFSEnabled = Configuration.global()
+    mClientWriteToUFSEnabled = mConf
         .getBoolean(PropertyKey.CLIENT_WRITE_TO_UFS_ENABLED);
   }
 
+  @VisibleForTesting
+  protected UnderFileSystem getUfsInstance(String ufsUriStr) {
+    AlluxioURI ufsUriUri = new AlluxioURI(ufsUriStr);
+    try {
+      Optional<UnderFileSystem> ufs = mUfsManager.get(ufsUriUri,
+          // todo(bowen): local configuration may not have UFS-specific configurations
+          //  find another way to load UFS configurations
+          UnderFileSystemConfiguration.defaults(mConf));
```

Comment:
specifically for hdfs, different hdfs might have different core-site / hdfs-site for us to pass in right? just asking

Reply:
Yes. Different HDFS can have different URI, configurations (from different `core-site.xml` conf files, etc.) or even different versions. We used to support all these but currently it lacks a way to get proper configuration inside worker.

Reply:
yeah actually no need to worry about that for HDFS, different nameservices hdfs-site could be aggregated into one file and and passed into Configuration